beater/cache.go

```diff
@@ -0,0 +1,89 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package beater
+
+import (
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/pkg/errors"
+	"golang.org/x/time/rate"
+)
+
+const burstMultiplier = 5
+
+var limiterPool sync.Pool
```

Comment:
this pool should probably be part of `rlCache`. Otherwise, you can create two `rlCache` with different limits who will use the same pool. That means the pool can contain rate limiters with different limits and you might end up calling `getRateLimiter` and getting a rate limit with the wrong limit.